### ISSUE TITLE
master does not compile on macOS without explicit ./configure --without-cart

### ISSUE THREAD

**Issue by glennklockwood** [hpc/ior#272]
There's something in `configure.ac` that is not properly setting `--without-cart` (and therefore includes Linux-specific CART build bits) on MacOS:

```
...
  CC       libaiori_a-utilities.o
  CC       libaiori_a-parse_options.o
  CC       libaiori_a-ior-output.o
  CC       libaiori_a-md-workbench.o
  CC       libaiori_a-aiori-DUMMY.o
  CC       libaiori_a-aiori-MPIIO.o
  CC       libaiori_a-aiori-MMAP.o
  CC       libaiori_a-aiori-POSIX.o
  AR       libaiori.a
  CCLD     ior
ld: warning: directory not found for option '-L/lib64'
ld: warning: directory not found for option '-L/lib'
ld: unknown option: --enable-new-dtags
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [ior] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all] Error 2
make: *** [all-recursive] Error 1
```

**Comment by glennklockwood** [comment]
This affects 3.3.0 and is blocking release.

**Comment by glennklockwood** [comment]
Fixed by 1e69bc2.
